homeassistant/components/history_stats/sensor.py

```diff
@@ -101,6 +102,9 @@ async def async_setup_platform(
 
     history_stats = HistoryStats(hass, entity_id, entity_states, start, end, duration)
     coordinator = HistoryStatsUpdateCoordinator(hass, history_stats, name)
+    await coordinator.async_refresh()
+    if not coordinator.last_update_success:
+        raise PlatformNotReady from coordinator.last_exception
```

Comment:
lol `async_setup_platform` 🙃

Reply:
This one took way longer to get it to this point as I spent too much time tracking down bugs which turned out to be sqlalchemy 🙈